optaplanner-core/src/main/java/org/optaplanner/core/impl/solver/AbstractSolver.java

```diff
@@ -114,6 +111,54 @@ public void solvingEnded(SolverScope<Solution_> solverScope) {
         phaseLifecycleSupport.fireSolvingEnded(solverScope);
     }
 
+    /**
+     * Used by a phase to notify the solver of its start.
+     * The event should not propagate any further, as that is the job of the phase.
+     *
+     * @param phaseScope never null
+     */
+    public void phaseStarted(AbstractPhaseScope<Solution_> phaseScope) {
+        bestSolutionRecaller.phaseStarted(phaseScope);
+        phaseLifecycleSupport.firePhaseStarted(phaseScope);
+        termination.phaseStarted(phaseScope);
```

Comment:
(important) Order, shouldn't termination be called before phaseLifecycleSupport? See original code.

Given that phaseLifecycleSupport is abstract/unknown, it's unlikely that termination would depend on anything in there. But visa versa, it's far more likely that that depends on termination.

Reply:
Applies to other 3 methods too.